### PR TITLE
Mh/tbbranch

### PR DESCRIPTION
**Pull request by holmesm8**
@caachz @agallant121 @TravisBorgsmiller 

This is the API consumption of the Sinatra Microservice.

### CODE REVIEW

File: Gemfile [hunk 1]
```diff
@@ -30,6 +30,7 @@ gem 'bootsnap', '>= 1.4.2', require: false
 gem 'rspec-rails'
 gem 'omniauth-google-oauth2'
 gem 'figaro'
+gem 'materialize-sass'
```
Comment (agallant121) [inline Gemfile:33]: Gem needed for materialize usage

File: app/controllers/dashboard_controller.rb [hunk 2]
```diff
@@ -1,5 +1,9 @@
 class DashboardController < ApplicationController
   before_action :authenticate
   def show
+    geolocation = GoogleGeolocationService.new
+    current_user.lat = geolocation.coordinates[:location][:lat]
+    current_user.lng = geolocation.coordinates[:location][:lng]
+    current_user.save!
```
Comment (caachz) [inline app/controllers/dashboard_controller.rb:7]: Thank you for adding that save :D

File: app/controllers/dashboard_controller.rb [hunk 3]
```diff
@@ -1,5 +1,9 @@
 class DashboardController < ApplicationController
   before_action :authenticate
   def show
+    geolocation = GoogleGeolocationService.new
+    current_user.lat = geolocation.coordinates[:location][:lat]
```
Comment (agallant121) [inline app/controllers/dashboard_controller.rb:5]: Uses the user's current lat and long that refreshes each time they visit the dashboard show page.

File: app/facades/rev_it_up_facade.rb [hunk 4]
```diff
@@ -0,0 +1,27 @@
+class RevItUpFacade
```
Comment (caachz) [inline app/facades/rev_it_up_facade.rb:1]: This facade looks great and is what Brian was showing us in the refactor so good job :D

File: app/controllers/restaurant_controller.rb [hunk 5]
```diff
@@ -0,0 +1,11 @@
+class RestaurantController < ApplicationController
+
+  def show
+    service = RevItUpService.new(current_user.lat, current_user.lng, meters_to_miles, params["price"])
```
Comment (agallant121) [inline app/controllers/restaurant_controller.rb:4]: Action to render the random restaurant based off the information from our sinatra, rev-it-up application.

Comment (caachz) [inline app/controllers/restaurant_controller.rb:4]: Refactor: We could have the service initialized within the facade to push this logic out of the view.

File: app/controllers/restaurant_controller.rb [hunk 6]
```diff
@@ -0,0 +1,11 @@
...
+    @restaurant = RevItUpFacade.new(service.restaurant)
+  end
+
+  def meters_to_miles
+    (params["distance"][0].to_i * 1609).to_i
```
Comment (agallant121) [inline app/controllers/restaurant_controller.rb:9]: Converter to switch the API's use of meters to miles for common use in the states.

File: app/views/dashboard/index.html.erb [hunk 7]
```diff
@@ -0,0 +1,13 @@
+Use My Current Location:<br>
+
+Travel Distance:<br>
+<%= text_field(:post, :radius) %> Miles<br><br><br>
+
+
+Price Range:<br>
+
+<% form_for @entry do |f| %>
```
Comment (caachz) [inline app/views/dashboard/index.html.erb:9]: what is this form for?

File: app/facades/rev_it_up_facade.rb [hunk 8]
```diff
@@ -0,0 +1,27 @@
...
+
+  def address
+    @data[:location][:display_address].join(" ")
+  end
+
+  def price
+    @data[:price]
+  end
+
+  def category
```
Comment (agallant121) [inline app/facades/rev_it_up_facade.rb:24]: Added category for the food choice shown on the user's dashboard after the shake it up has happened.

File: app/helpers/application_helper.rb [hunk 9]
```diff
@@ -1,2 +1,19 @@
 module ApplicationHelper
+  def distance
+    # position 0 is the displayed text and position 1 is the tag
+    [
+      ['1 Mile', '1 Mile'],
```
Comment (agallant121) [inline app/helpers/application_helper.rb:5]: These numbers will conflict with current branch that has been pulled down and will need to be dealt with before a merge.

File: spec/features/gets_user_geolocation_information_spec.rb [hunk 10]
```diff
@@ -0,0 +1,11 @@
+require 'rails_helper'
+
+RSpec.describe 'As a user' do
```
Comment (caachz) [inline spec/features/gets_user_geolocation_information_spec.rb:3]: need to finish this test

File: spec/features/user_can_see_random_restaurant_info_spec.rb [hunk 11]
```diff
@@ -0,0 +1,29 @@
...
+      expect(page).to have_content('Gilmore')
+    end
+    it 'shows me a random restaurant with info' do
+
+      visit '/dashboard'
+      select "1 Mile", from: :distance
+      select "$$$", from: :price
+      click_button 'Shake It Up!'
+
+      within '#restaurant' do
```
Comment (caachz) [inline spec/features/user_can_see_random_restaurant_info_spec.rb:21]: good test :D

File: spec/helpers/home_helper_spec.rb [hunk 12]
```diff
@@ -1,15 +1,15 @@
...
+# #   describe "string concat" do
+# #     it "concats two strings with spaces" do
+# #       expect(helper.concat_strings("this","that")).to eq("this that")
+# #     end
+# #   end
+# # end
+# RSpec.describe HomeHelper, type: :helper do
+#   pending "add some examples to (or delete) #{__FILE__}"
 # end
-RSpec.describe HomeHelper, type: :helper do
```
Comment (caachz) [inline spec/helpers/home_helper_spec.rb:13]: thank you for getting rid of homehelper

File: spec/rails_helper.rb [hunk 13]
```diff
@@ -1,5 +1,10 @@
 # This file is copied to spec/ when you run 'rails generate rspec:install'
+require 'simplecov'
```
Comment (caachz) [inline spec/rails_helper.rb:2]: great :D now we have simplecov

File: app/views/dashboard/index.html.erb (hunk 7 above)
Comment (agallant121) [inline app/views/dashboard/index.html.erb:9]: Potentially going to be removed as this functionality has been replaced already.

File: app/views/layouts/application.html.erb [hunk 14]
```diff
@@ -10,6 +10,16 @@
   </head>
 
   <body>
+    <div class="container">
+      <nav>
+        <div class="nav-wrapper blue-grey darken-1">
+          <a href="#" class="brand-logo center">ShakeItUp</a>
+          <% if current_user != nil %>
```
Comment (agallant121) [inline app/views/layouts/application.html.erb:17]: Current user could be just 'if current_user' without the !=

File: config/initializers/omniauth.rb [hunk 15]
```diff
@@ -1,3 +1,3 @@
 Rails.application.config.middleware.use OmniAuth::Builder do
-  provider :google_oauth2, ENV['GOOGLE_CLIENT_ID'], ENV['GOOGLE_CLIENT_SECRET']
+  provider :google_oauth2, ENV['GOOGLE_CLIENT_ID'], ENV['GOOGLE_CLIENT_SECRET'], :skip_jwt => true
```
Comment (agallant121) [inline config/initializers/omniauth.rb:2]: skip at the end is necessary to work

File: config/routes.rb [hunk 16]
```diff
@@ -1,11 +1,13 @@
...
-get 'login', to: redirect('/auth/google_oauth2'), as: 'login'
-get 'logout', to: 'sessions#destroy', as: 'logout'
-get 'auth/:provider/callback', to: 'sessions#create'
-get 'auth/failure', to: redirect('/')
-get 'home', to: 'home#show'
-get '/dashboard', to: 'dashboard#show', as: 'dashboard'
-#change me to dashboard at some point and can be the page the user uses to shake it up
-root to: "home#show"
+  # get '/', to: 'welcome#index'
+  # get '/dashboard', to: 'dashboard#index'
```
Comment (agallant121) [inline config/routes.rb:4]: Removed overlapping routes that were named different things in development.

File: config/webpack/environment.js [hunk 17]
```diff
@@ -1,3 +1,10 @@
 const { environment } = require('@rails/webpacker')
 
+const webpack = require('webpack')
+environment.plugins.append('Provide', new webpack.ProvidePlugin({
+  $: 'jquery',
+  jQuery: 'jquery',
```
Comment (agallant121) [inline config/webpack/environment.js:6]: This was needed to render the drop down menus.

**Pull request closed without merging**